docs: update gateway README (#10368)
emqx/emqx: apps/emqx_gateway/README.md

# Gateway

> EMQX Gateway is an application that managing all gateways in EMQX.

Reviewer: ```suggestion
EMQX Gateway is an application framework that manages all gateways within EMQX.
```

Author: Great thanks!!! @sstrigler

- Configuration & Schema
- HTTP/CLI management interfaces

> The emqx_gateway application depends on `emqx`, `emqx_authn`, `emqx_ctl` that

Author: ```suggestion
The emqx_gateway application depends on `emqx`, `emqx_authn`, `emqx_authz`, `emqx_ctl` that
```

> provide the foundation for protocol access.

Reviewer 2: and also `emqx_authz`
from:
```
apps/emqx_gateway/src/emqx_gateway_ctx.erl
50:    authorize/4
157:-spec authorize(
164:authorize(_Ctx, ClientInfo, PubSub, Topic) ->
165:    emqx_access_control:authorize(ClientInfo, PubSub, Topic).
```

More introduction: [Extended Protocol Gateway](https://www.emqx.io/docs/en/v5.0/gateway/gateway.html)

## Usage

This application is just a Framework, we provide some standard implementations,
such as [Stomp](../emqx_stomp/README.md), [MQTT-SN](../emqx_mqttsn/README.md),
[CoAP](../emqx_coap/README.md) and [LwM2M](../emqx_lwm2m/README.md) gateway.

These applications are all packaged by default in the EMQX distribution. If you
need to start a certain gateway, you only need to enable it via
Dashboard, HTTP API or emqx.conf file.

For instance, enable the Stomp gateway in emqx.conf:
```hocon
gateway.stomp {

  mountpoint = "stomp/"

  listeners.tcp.default {
    bind = 61613
    acceptors = 16
    max_connections = 1024000
    max_conn_rate = 1000
  }
}
```

## How to develop your Gateway application

> There are three ways to develop Gateway application to accept your private protocol

Reviewer: ```suggestion
There are three ways to develop a Gateway application to accept your private protocol
```

Refer to: [ExProto Gateway](../emqx_exproto/README.md)

> ## Understard emqx_gateway framework

Reviewer 3: ```suggestion
## Understand emqx_gateway framework
```
